### PR TITLE
GEODE-8219: Apply RedisResponse to all redis string commands

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/SetBitExecutor.java [hunk 1]
```diff
@@ -81,31 +79,30 @@ public void executeCommand(Command command, ExecutionHandlerContext context) {
       }
       stringCommands.set(key, new ByteArrayWrapper(bytes), null);
       command.setResponse(Coder.getIntegerResponse(context.getByteBufAllocator(), 0));
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/SetBitExecutor.java:81]: I think this setReponse call should have been deleted

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/BitPosExecutor.java [hunk 2]
```diff
@@ -47,27 +48,17 @@ public void executeCommand(Command command, ExecutionHandlerContext context) {
...
-      // key
-      // does
-      // not
-      // exists
-      // for
-      // this
-      // command
-      return;
+      // Redis returns 0 when key does not exists for this command
+      command.setResponse(Coder.getIntegerResponse(context.getByteBufAllocator(), -bit));
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/BitPosExecutor.java:60]: Should this setResponse be removed?

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/SetBitExecutor.java [hunk 3]
```diff
@@ -34,12 +35,13 @@
...
   @Override
-  public void executeCommand(Command command, ExecutionHandlerContext context) {
+  public RedisResponse executeCommandWithResponse(Command command,
+      ExecutionHandlerContext context) {
     List<byte[]> commandElems = command.getProcessedCommand();
 
     if (commandElems.size() < 4) {
       command.setResponse(Coder.getErrorResponse(context.getByteBufAllocator(), ArityDef.SETBIT));
-      return;
+      return RedisResponse.error(ArityDef.SETBIT);
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/SetBitExecutor.java:44]: remove this setResponse calls

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/SetExecutor.java [hunk 4]
```diff
@@ -42,23 +44,24 @@ public void executeCommand(Command command, ExecutionHandlerContext context) {
...
+      ByteArrayWrapper key,
       ByteArrayWrapper value, RedisStringCommands redisStringCommands, SetOptions setOptions) {
 
     boolean result = redisStringCommands.set(key, value, setOptions);
 
     if (result) {
       command.setResponse(Coder.getSimpleStringResponse(context.getByteBufAllocator(), SUCCESS));
-    } else {
-      command.setResponse(Coder.getNilResponse(context.getByteBufAllocator()));
+      return RedisResponse.string(SUCCESS);
```
Comment (dschneider-pivotal) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/SetExecutor.java:61]: remove this setResponse call

File: geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/SetBitExecutor.java (hunk 1 above)
Comment (jdeppe-pivotal) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/executor/string/SetBitExecutor.java]: Doh! Thanks.

**Pull request merged**